src/server/api/routers/ballot.ts

```diff
@@ -175,16 +175,22 @@ export const ballotRouter = createTRPCRouter({
               })),
           );
         } else if (ctx.round?.type === "impact") {
-          return ballots.flatMap(
-            ({ voterId, signature, publishedAt, allocations }) =>
-              allocations.map(({ amount, id }) => ({
-                voterId,
-                signature,
-                publishedAt,
-                amount,
-                id,
-              })),
-          );
+          const allocations = await ctx.db.allocation.findMany({
+            where: {
+              roundId: ctx.round.id,
+              ballot: { publishedAt: { not: null } },
+            },
+            include: {
+              ballot: true,
+            },
+          });
+          return allocations.map(({ voterId, ballot: { signature, publishedAt }, amount, id }) => ({
+            voterId,
+            signature,
+            publishedAt,
+            amount,
+            id,
+          }));
```

Comment:
We might want to add the ballot type here also:
```tsx
ballot: { publishedAt: { not: null }, type: "impact" },
```

I know it's unlikely to query allocations in ballots with type `project` because the round type would need to be changed after voting has started. Now thinking about it, updating round type once there are allocations should not be possible. 

Then we could probably remove the `ballot.type`.

Reply:
Yeah i thought the same ! Once the round starts, it won't change type cause that would have side-effect which would break a lot of things. Hence left the filter out